lib/date_time/date_time.c

```diff
@@ -479,53 +143,42 @@ static void new_date_time_get(void)
 
 		LOG_DBG("Updating date time UTC...");
 
-		err = current_time_check();
+		err = date_time_current_check();
 		if (err == 0) {
 			LOG_DBG("Using previously obtained time");
-			initial_valid_time = true;
 			date_time_schedule_update();
-			date_time_notify_event(&evt);
+			date_time_notify_event(DATE_TIME_EVT_TYPE_PREVIOUS);
 			continue;
 		}
 
-		LOG_DBG("Current time not valid");
-
 #if defined(CONFIG_DATE_TIME_MODEM)
 		LOG_DBG("Getting time from cellular network");
+		int64_t date_time_ms_modem = 0;
 
-		err = time_modem_get();
+		err = date_time_modem_get(&date_time_ms_modem);
 		if (err == 0) {
-			initial_valid_time = true;
-			evt.type = DATE_TIME_OBTAINED_MODEM;
-			date_time_store(time_aux.date_time_utc);
-			date_time_schedule_update();
-			date_time_notify_event(&evt);
+			date_time_store(date_time_ms_modem, DATE_TIME_OBTAINED_MODEM);
 			continue;
 		}
 #endif
 #if defined(CONFIG_DATE_TIME_NTP)
 		LOG_DBG("Getting time from NTP server");
+		int64_t date_time_ms_ntp = 0;
 
-		err = time_NTP_server_get();
+		err = date_time_ntp_get(&date_time_ms_ntp);
 		if (err == 0) {
-			initial_valid_time = true;
-			evt.type = DATE_TIME_OBTAINED_NTP;
-			date_time_store(time_aux.date_time_utc);
-			date_time_schedule_update();
-			date_time_notify_event(&evt);
+			date_time_store(date_time_ms_ntp, DATE_TIME_OBTAINED_NTP);
 			continue;
 		}
 #endif
```

Comment:
This could be done with a for-loop if we have modem and NTP methods as a structure with function pointers similarly to location methods in Location library (location_core.c). We could do something more elegant with that solution in few places but I didn't do that before hearing opinions (@simensrostad) as it might be also over-engineering in this case when we probably don't expect more methods.

Reply:
I don't have any opinion here really. If you think it will improve the library and you have the time, why not? I'll leave that up to you 😄 I don't expect any methods to be added any time soon. If users want to add time from external sources, they can always use `date_time_set`.

Reply:
Not going to do any major changes for this anymore. It would be a bit of over-engineering in this case...